interface/main/calendar/add_edit_event.php

```diff
@@ -1336,7 +1336,15 @@ function find_available(extra) {
         <?php endif ?>
     </ul>
 </nav> <!-- nav-group -->
-<form role="form" method='post' name='theform' id='theform' action='add_edit_event.php?eid=<?php echo attr($eid) ?>'>
+<?php
+$form_id = 'theform';
+if (!empty($_GET['prov']) && ($_GET['prov'] == true)) {
+    $form_id = 'theform_prov';
+} elseif ($_GET['group'] == true) {
+    $form_id = 'theform_groups';
+}
+?>
+<form role="form" method='post' name='<?php echo $form_id; ?>' id='<?php echo $form_id; ?>' action='add_edit_event.php?eid=<?php echo attr_url($eid) ?>'>
```

Comment:
To be super safe would wrap both $form_id echos above with attr() function to htmlescape. May be seen as overkill since the variable is hard-coded above, however is somebody in the future decides to modify it and incorporate a user defined variable in it, then the attr() will stop a xss vulnerability in that case. For this reason, we generally just escape everything getting echos without putting much thought into where it comes from (except in the special case where it will include html in it, then need to escape further upstream in the code). Hope this makes sense.